mitodl/ocw-studio
# Add API for WebsiteContent

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of #52

#### What's this PR do?
Adds a website content REST API, a prerequisite to interact with website data 

#### How should this be manually tested?

 - View the list API for website content which should look something like this: `/api/websites/course-id-here/content/`. You should see a smallish list of objects with only a `uuid`, `type`, and `title`. The count should match the number of `WebsiteContent` objects referencing that `Website` with a foreign key. 
 - At the same API, create a new document. You should see a 201 result with the content you specified included in the result. You should see it in the list view as well. A valid `type `is required but you can optionally leave out `title` and `markdown` -- they will be `null` but you can `PATCH` them later. You won't be able to `PATCH` `type` so it needs to be set correctly on create.
 - Go to `/api/websites/course-id-here/content/uuid-for-some-content/` to view the detail view, using one of the uuids from the list view. You should see the same information plus markdown, which was omitted from the list view to reduce data. You should be able to `PATCH` the markdown and title and see the results persist. If you try to patch `type` or `uuid` it should ignore the change.

You should not see any option (at least in this PR) to `DELETE` a `WebsiteContent`.
